certora/specs/SocialRecoveryModule.spec

```diff
@@ -502,3 +503,119 @@ rule finalizeRecoveryAlwaysPossible(env e) {
+    f(e, args);
+
+    bool confirmedHashAfter = currentContract.confirmedHashes[hash][guardian];
+
+    assert confirmedHashBefore != confirmedHashAfter =>
+        f.selector == sig:confirmRecovery(address,address[],uint256,bool).selector ||
+        f.selector == sig:multiConfirmRecovery(address,address[],uint256,SocialRecoveryModule.SignatureData[],bool).selector;
+}
+
+// This rule verifies that is the walletsNonces change, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+// - executeRecovery(...)
+// - invalidateNonce(...)
+rule walletsNoncesChange(method f) {
+    uint256 walletsNoncesBefore = currentContract.walletsNonces[safeContract];
+
+    env e;
+    calldataarg args;
+    f(e, args);
+
+    uint256 walletsNoncesAfter = currentContract.walletsNonces[safeContract];
+
+    assert walletsNoncesBefore != walletsNoncesAfter =>
+        f.selector == sig:confirmRecovery(address,address[],uint256,bool).selector ||
+        f.selector == sig:multiConfirmRecovery(address,address[],uint256,SocialRecoveryModule.SignatureData[],bool).selector ||
+        f.selector == sig:executeRecovery(address,address[],uint256).selector ||
+        f.selector == sig:invalidateNonce().selector;
+}
+
+// This rule verifies that the recovery period never changes.
+rule recoveryPeriodNeverChange(method f) {
+    uint256 recoveryPeriodBefore = currentContract.recoveryPeriod;
+
+    env e;
+    calldataarg args;
+    f(e, args);
+
+    uint256 recoveryPeriodAfter = currentContract.recoveryPeriod;
+
+    assert recoveryPeriodBefore == recoveryPeriodAfter;
+}
+
+// This rule verifies that the guardian data can only be changed by the following functions:
+// - addGuardianWithThreshold(...)
+// - revokeGuardianWithThreshold(...)
+// - changeThreshold(...)
+rule entriesChange(method f) {
+    bytes32 guardiansHashBefore = currentContract.guardiansHash(safeContract);
+    uint256 guardiansCountBefore = currentContract.guardiansCount(safeContract);
+    uint256 guardianThresholdBefore = currentContract.threshold(safeContract);
+
+    env e;
+    calldataarg args;
+    f(e, args);
+
+    bytes32 guardiansHashAfter = currentContract.guardiansHash(safeContract);
+    uint256 guardiansCountAfter = currentContract.guardiansCount(safeContract);
+    uint256 guardianThresholdAfter = currentContract.threshold(safeContract);
+
+    assert (
+        guardiansHashBefore != guardiansHashAfter ||
+        guardiansCountBefore != guardiansCountAfter
+    ) =>
+        f.selector == sig:addGuardianWithThreshold(address,uint256).selector ||
+        f.selector == sig:revokeGuardianWithThreshold(address,address,uint256).selector;
+    assert guardianThresholdBefore != guardianThresholdAfter =>
+        f.selector == sig:addGuardianWithThreshold(address,uint256).selector ||
+        f.selector == sig:revokeGuardianWithThreshold(address,address,uint256).selector ||
+        f.selector == sig:changeThreshold(uint256).selector;
```

Comment:
nit: I feel like instead of two asserts this can be split into multiple rules for easier readability

Reply:
Just to clarify: You mean different rules for:
- guardians, count and threshold OR
- one rule for guardians and count, and another for threshold?

Reply:
the second

Reply:
Done 👍🏾